frontend/src/components/GifWidget/useGifWidget.tsx

```diff
@@ -0,0 +1,61 @@
+import axios from "axios";
+import { GifListImage } from "components/GifWidget/GifWidget.styled";
+import { ChangeEvent, useEffect, useRef, useState } from "react";
+import { useOnClickOutside } from "usehooks-ts";
+
+interface IGifList {
+    id: string;
+    images: Record<
+        string,
+        {
+            url: string;
+        }
+    >;
+}
+
+const useGifWidget = (toggleGifWidgetVisibility: () => void) => {
+    const [gifsIsLoading, setGifsIsLoading] = useState(true);
+    const [gifList, setGifList] = useState<IGifList[]>([]);
+    const [gifSearchInput, setGifSearchInput] = useState("");
+
+    const ref = useRef(null);
+
+    const findGif = (event: ChangeEvent<HTMLInputElement>) => {
+        setGifSearchInput(event.target.value);
+        setGifsIsLoading(true);
+    };
+
+    const fetchGifs = () =>
+        gifList.map(item => (
+            <GifListImage
+                key={item.id}
+                src={item.images.fixed_height_downsampled.url}
+                onClick={() => console.log()} //item.images.fixed_height.url
```

Comment:
Remove it

Reply:
That's for save, because in the future implementation, i don't need to search it.